debian/postrm

```diff
@@ -19,8 +19,6 @@ remove_logs(){
 remove_esm(){
     # config files created in postinst, need explicit handling on purge
     rm -f /etc/apt/trusted.gpg.d/ubuntu-advantage-esm-infra-trusty.gpg
-    rm -f /etc/apt/sources.list.d/ubuntu-esm-infra-trusty.list
-    rm -f /etc/apt/preferences.d/ubuntu-esm-infra-trusty
```

Comment:
My former comment was completely incorrect. 

remove_esm should either be renamed remove_postinst_created_files.  It is only there to remove any untracked package files that were created by debian/postinst (not necessarily due to ua enable <service>).
Trusty is the only case where this happens because ua-tools pkg pre-enables visibility to esm files (unauthenticated).

So, these removes should still be here. All other apt files are removed automatically due in debian/prerm calling clean_apt_files().

Also there is a bug with prerm #1170 that needs fixing separately

```suggestion
    rm -f /etc/apt/sources.list.d/ubuntu-esm-infra-trusty.list
    rm -f /etc/apt/preferences.d/ubuntu-esm-infra-trusty
```